include/deal.II/grid/reference_cell.h

```diff
@@ -2172,11 +2172,46 @@ ReferenceCell::contains_point(const Point<dim> &p, const double tolerance) const
     }
   else if (*this == ReferenceCells::Wedge)
     {
-      Assert(false, ExcNotImplemented());
+      // The wedge we use is a triangle extruded into the third
+      // dimension by one unit. So we can use the same logic as for
+      // triangles above (i.e., for the simplex above, using dim==2)
+      // and then check the third dimension separately.
+
+      for (unsigned int d = 0; d < 2; ++d)
+        if (p[d] < -tolerance)
+          return false;
+
+      const double sum = p[0] + p[1];
+      if (sum > 1 + tolerance * std::sqrt(2.0))
+        return false;
+
+      if (p[2] < -tolerance)
+        return false;
+      if (p[2] > 1 + tolerance)
+        return false;
+
+      return true;
     }
   else if (*this == ReferenceCells::Pyramid)
     {
-      Assert(false, ExcNotImplemented());
+      // A pyramid only lives in the upper half-space:
+      if (p[2] < -tolerance)
+        return false;
+
+      // It also only lives in the space below z=1:
+      if (p[2] > 1 + tolerance)
+        return false;
```

Comment:
Looks like you need something like 
```suggestion
      if (p[dim < 3 ? 0 : 2] < -tolerance)
        return false;

      // It also only lives in the space below z=1:
      if (p[dim < 3 ? 0 : 2] > 1 + tolerance)
        return false;
```
to make the CI happy since we (still) can't use `if constexpr`.

Reply:
Ah, that's aggravating. Let's see what happens now.

Reply:
There are more places that need this annoying workaround, see https://github.com/dealii/dealii/actions/runs/3896751841/jobs/6653789602.

Reply:
I gave it a more aggressive try...